src/test/java/io/jenkins/plugins/checks/github/GitSCMChecksContextTest.java

```diff
@@ -0,0 +1,37 @@
+package io.jenkins.plugins.checks.github;
+
+import hudson.model.Run;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.UserRemoteConfig;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class GitSCMChecksContextTest {
+    @Test
+    public void shouldGetRepository() {
```

Comment:
could we extract the logic out to pass a string to a method so we don't have to do all this mocking?